packages/headless/src/database/mongodb/product/index.ts

```diff
@@ -49,4 +50,28 @@ export class ProductDatabase implements IProductDatabase {
   async getProductsList(skip: number, limit: number, query?: Record<string, any>, sortCondition?: string): Promise<Product[] | []> {
     return await ProductModel.find(query, '-_id').sort(sortCondition).skip(skip).limit(limit).lean();
   }
+
+  async GetTopSellingProducts(query: Record<string, any>, skip: number, limit: number): Promise<Product[] | []> {
+    const orderArray = await OrderModel.aggregate([
+      { $unwind: "$products" },
+      { $unwind: "$products.info" },
+      {
+        $group:
+        {
+          _id: "$products.id",
+          totalOrderQuantity:
```

Comment:
as the function returns the top selling products, there should be a threshold for totalOrderQuantity which will define the lower bound of "top". and also a date-time threshold (like last 7 days) can be added.